### PR TITLE
[wasi] CI: Add AOT library tests job

### PR DESCRIPTION
**Pull request by radical**
- misc fixes
- [wasi] Add support for building AOT library tests on helix
- [wasi] CI: Add AOT smoke tests
- Share targets for running AOT tests
- cleanup

This adds the job to `runtime-wasm` also, which will run a larger subset of tests but will ignore any failures. For the smoke tests job, failures are not ignored.

### CODE REVIEW

File: eng/testing/tests.browser.targets [hunk 1]
```diff
@@ -260,12 +260,14 @@
       <RunScriptCommands Include="if [[ &quot;$SCENARIO&quot; == &quot;WasmTestOnNodeJS&quot; || &quot;$SCENARIO&quot; == &quot;wasmtestonnodejs&quot; ]]; then npm ci; fi" />
 
       <SetScriptCommands Condition="'$(InstallChromeForTests)' == 'true' and '$(ChromeDriverBinaryPath)' != ''" Include="export PREPEND_PATH=$([System.IO.Path]::GetDirectoryName($(ChromeDriverBinaryPath)))" />
+      <SetScriptCommands Include="export RuntimeIdentifier=browser-wasm" />
```
Comment (radical) [inline eng/testing/tests.browser.targets:263]: note to self: this is why some wbt tests are failing

**Comment by radical** [comment]
Failures are unrelated.

File: eng/testing/tests.wasi.targets [hunk 2]
```diff
@@ -43,7 +49,7 @@
     <_XHarnessArgs Condition="'$(WasmXHarnessArgs)' != ''"         >$(_XHarnessArgs) $(WasmXHarnessArgs)</_XHarnessArgs>
     <_XHarnessArgs Condition="'$(WasmXHarnessTestsTimeout)' != ''" >$(_XHarnessArgs) &quot;--timeout=$(WasmXHarnessTestsTimeout)&quot;</_XHarnessArgs>
     <_XHarnessArgs Condition="'$(WasmXHarnessArgsCli)' != ''"      >$(_XHarnessArgs) $(WasmXHarnessArgsCli)</_XHarnessArgs>
-    
+
```
Comment (vargaz) [inline eng/testing/tests.wasi.targets:52]: Adding
```<_XHarnessArgs>$(_XHarnessArgs) --engine-arg=--max-wasm-stack=134217728</_XHarnessArgs>```
fixes stack overflows in the System.Runtime.Serialization.Xml tests.

Comment (radical) [reply eng/testing/tests.wasi.targets:52]: I can add this in a follow up PR.

Comment (radical) [reply eng/testing/tests.wasi.targets:52]: Should this be used for all the aot tests, or only for `System.Runtime.Serialization.Xml`?

Comment (radical) [reply eng/testing/tests.wasi.targets:52]: Also, will this help with oom in System.Globalization.Tests too? https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-95146-merge-7b4bd653d93042109e/System.Globalization.Tests/1/console.cb452bd4.log?helixlogtype=result

Comment (vargaz) [reply eng/testing/tests.wasi.targets:53]: It should be used for all tests probably. The OOM happens during the aot build, so this will not fix it.

**Review by lewing — APPROVED** [review]
Please open an issue for the unflagged debugger timeout before merging

File: src/mono/wasi/build/WasiApp.props [hunk 3]
```diff
@@ -3,7 +3,7 @@
     <!-- FIXME: update the condition for this in Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets -->
     <_RuntimeIdentifierUsesAppHost>false</_RuntimeIdentifierUsesAppHost>
     <TargetArchitecture>wasm</TargetArchitecture>
-    <TargetOS>Browser</TargetOS>
+    <TargetOS>wasi</TargetOS>
```
Comment (lewing) [inline src/mono/wasi/build/WasiApp.props:6]: ah... this is where that was coming from.

**Comment by radical** [comment]
> Please open an issue for the unflagged debugger timeout before merging

`Build Analysis` is still showing the timeout but the tests were re-run automatically, and didn't fail. I have opened https://github.com/dotnet/runtime/issues/95343 anyway.

**Pull request merged**